test/controllers/search_controller_test.rb

```diff
@@ -455,4 +455,391 @@ def source_filter_count(controller)
+
+        assert_select 'input#geobox-minLongitude', value: 40.5
+        assert_select 'input#geobox-minLatitude', value: 60.0
+        assert_select 'input#geobox-maxLongitude', value: 78.2
+        assert_select 'input#geobox-maxLatitude', value: 80.0
+        assert_select 'input#geodistance-latitude', value: 36.1
+        assert_select 'input#geodistance-longitude', value: 62.6
+        assert_select 'input#geodistance-distance', value: '50mi'
+      end
+    end
+
+    test 'geospatial search can combine with basic and advanced inputs' do
+      VCR.use_cassette('geo all',
+                        allow_playback_repeats: true,
+                        match_requests_on: %i[method uri body]) do
+        query = {
+          q: 'hi',
+          title: 'hey',
+          citation: 'hello',
+          geobox: 'true',
+          geodistance: 'true',
+          geoboxMinLongitude: 40.5,
+          geoboxMinLatitude: 60.0,
+          geoboxMaxLongitude: 78.2,
+          geoboxMaxLatitude: 80.0,
+          geodistanceLatitude: 36.1,
+          geodistanceLongitude: 62.6,
+          geodistanceDistance: '50mi' 
+        }.to_query
+        get "/results?#{query}"
+        assert_response :success
+        assert_nil flash[:error]
+      end
+    end
+
+    test 'geobox arguments are non-nullable' do
+      query = {
+        geobox: 'true'
+      }.to_query
+      get "/results?#{query}"
+      assert_response :redirect
+      assert_equal flash[:error], "All bounding box fields are required."
+    end
+
+    test 'geodistance arguments are non-nullable' do
+      query = {
+        geodistance: 'true'
+      }.to_query
+      get "/results?#{query}"
+      assert_response :redirect
+      assert_equal flash[:error], "All geospatial distance fields are required."
+    end
+
+    test 'geobox param is required for geobox search' do
+      query = {
+        geoboxMinLongitude: 40.5,
+        geoboxMinLatitude: 60.0,
+        geoboxMaxLongitude: 78.2,
+        geoboxMaxLatitude: 80.0
+      }.to_query
+      get "/results?#{query}"
+      assert_response :redirect
+      assert_equal flash[:error], "A search term is required."
+    end
+
+    test 'geodistance param is required for geobox search' do
+      query = {
+        geodistanceLatitude: 36.1,
+        geodistanceLongitude: 62.6,
+        geodistanceDistance: '50mi' 
+      }.to_query
+      get "/results?#{query}"
+      assert_response :redirect
+      assert_equal flash[:error], "A search term is required."
+    end
+
+    test 'geobox lat/long cannot fall below min limit' do
+      low_query = {
+        geobox: 'true',
+        geoboxMinLongitude: -180.2,
+        geoboxMinLatitude: -90.2,
+        geoboxMaxLongitude: -180.1,
+        geoboxMaxLatitude: -90.1
+      }.to_query
+      get "/results?#{low_query}"
+      assert_response :redirect
+      assert_equal flash[:error], "Latitude must be between -90.0 and 90.0, and longitude must be -180.0 and 180.0."
```

Comment:
Something is odd about this test, but I'm having a hard time putting my finger on it. I see that response when I try the query in the UI, but when I try submitting individual out-of-limit values I get different results.

This query works, as it should (although the bounding box is non-sensical, near the south pole)
https://timdex-ui-pi-gdt-136-lo-2ovnii.herokuapp.com/results?q=&geobox=true&geoboxMinLongitude=-180&geoboxMinLatitude=-90&geoboxMaxLongitude=-179&geoboxMaxLatitude=-89

The next two searches each violate only the minimum lat or long parameters, and each seems to respond  with a 500 error from the API, not a flash message:
* minLongitude too low
https://timdex-ui-pi-gdt-136-lo-2ovnii.herokuapp.com/results?q=&geobox=true&geoboxMinLongitude=-180.2&geoboxMinLatitude=-90&geoboxMaxLongitude=-179&geoboxMaxLatitude=-89

* minLatitude too low
https://timdex-ui-pi-gdt-136-lo-2ovnii.herokuapp.com/results?q=&geobox=true&geoboxMinLongitude=-180&geoboxMinLatitude=-90.2&geoboxMaxLongitude=-179&geoboxMaxLatitude=-89

The next two searches violate only the _maximum_ lat or long parameters, and appear to be tripped up by the bounding box checks for whether min and max values are in the right order? The flash message reads "Bounding box minimum lat/long cannot exceed maximum lat/long"
* maxLongitude too low
https://timdex-ui-pi-gdt-136-lo-2ovnii.herokuapp.com/results?q=&geobox=true&geoboxMinLongitude=-180&geoboxMinLatitude=-90&geoboxMaxLongitude=-180.2&geoboxMaxLatitude=-90

* maxLatitude too low
https://timdex-ui-pi-gdt-136-lo-2ovnii.herokuapp.com/results?q=&geobox=true&geoboxMinLongitude=-180&geoboxMinLatitude=-90&geoboxMaxLongitude=-179&geoboxMaxLatitude=-90.2

As I read the search controller, it looks like the checks should be performed in the order of "presence, range, values" - so I would expect that the above geobox searches would trip the range validations first, and not reach the value validations - but that's not what is happening (and in some cases the invalid values are getting passed by every check, and the API is returning an error).

Looking at the geobox range validation again, I think there's at least one issue with the logic (which I've commented up there). I wonder whether the testing strategy here should be to send in one invalid value at a time, rather than a set of entirely invalid values?